### DISCUSSION TITLE
Is it correct to run Quartz Scheduler every time for new users?

### DISCUSSION THREAD

**Discussion by Pooriamoosavi** [quartznet/quartznet#584] [General]
Quartz Version 3.0.2 
I have some users which register in my system then for every single one of them I should save time and every week send the a message so these times are going to be different due to their registration date. Question is here that is it correct to run the scheduler every time?
my code is like this :
`class JobScheduler
    {
        public static StdSchedulerFactory stdScheduler = new StdSchedulerFactory();
        public static async void Start(DateTime registrationDate, string chatId)
        {
            IScheduler sc = await stdScheduler.GetScheduler();

            await sc.Clear();
            Console.WriteLine("Cleared!");

            ISimpleTrigger trigger = (ISimpleTrigger)TriggerBuilder.Create()
            .WithIdentity("LCP", "weeklyMessages")
            .ForJob("LCPJob", "WeeklyMassages")
            .StartAt(registrationDate)
            .Build();

            IJobDetail lcpjob = JobBuilder.CreateForAsync<LCPJob>()
            .WithIdentity("LCPJob", "WeeklyMassages")
            .UsingJobData("chatId", chatId)
             .Build();

            await sc.ScheduleJob(lcpjob, trigger);
            Console.WriteLine("First Trigger");

            ISimpleTrigger trigger2 = (ISimpleTrigger)TriggerBuilder.Create()
            .WithIdentity("RSS", "weeklyMessages")
            .StartAt(registrationDate)
            .Build();

            IJobDetail rssjob = JobBuilder.Create<RSSJob>()
            .WithIdentity("RSSJob", "wMessages")
            .UsingJobData("chatId", chatId)
            .Build();

            await sc.ScheduleJob(rssjob, trigger2);
            Console.WriteLine("Second Trigger");

            await sc.Start();

        }
    }`
 and I am calling this method for every single new member!

**Comment by Sicos1977** [comment]
It is easier to keep the scheduler running and just add a new job when needed.

**Comment by sridhargoshika** [comment]
@Pooriairoop any update on this issue?  I am calling start method for every single new member! But schedule jobs not executing after some time. It seems like destroying scheduled jobs even my application pool mode is set alwaysrunnining.

**Comment by lahma** [comment]
I would use jobs for specific tasks and then use triggers for each user / use case and name the trigger using some id that relates to actual user. Then you can check if the trigger exists or not and add/modify if necessary. You can add parameters to trigger which will be transferred to job execution (like target user id).
